example/particles/particles.cpp

```diff
@@ -0,0 +1,361 @@
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+
+#include "particles.hpp"
+
+#include <algorithm>
+#include <limits>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+// *************************************************//
+// redefine some internal parthenon functions      *//
+// *************************************************//
+namespace particles_example {
+
+Packages_t ProcessPackages(std::unique_ptr<ParameterInput> &pin) {
+  Packages_t packages;
+  packages["particles_package"] = particles_example::Particles::Initialize(pin.get());
+  return packages;
+}
+
+void ProblemGenerator(MeshBlock *pmb, ParameterInput *pin) {
+  // Don't do anything for now
+}
+
+// *************************************************//
+// define the "physics" package particles_package, *//
+// which includes defining various functions that  *//
+// control how parthenon functions and any tasks   *//
+// needed to implement the "physics"               *//
+// *************************************************//
+
+namespace Particles {
+
+std::shared_ptr<StateDescriptor> Initialize(ParameterInput *pin) {
+  auto pkg = std::make_shared<StateDescriptor>("particles_package");
+
+  int num_particles = pin->GetOrAddInteger("Particles", "num_particles", 100);
+  pkg->AddParam<>("num_particles", num_particles);
+  Real particle_speed = pin->GetOrAddReal("Particles", "particle_speed", 1.0);
+  pkg->AddParam<>("particle_speed", particle_speed);
+  Real const_dt = pin->GetOrAddReal("Particles", "const_dt", 1.0);
+  pkg->AddParam<>("const_dt", const_dt);
+
+  // Initialize random number generator pool
+  int rng_seed = pin->GetInteger("Particles", "rng_seed");
+  pkg->AddParam<>("rng_seed", rng_seed);
+  RNGPool rng_pool(rng_seed);
+  pkg->AddParam<>("rng_pool", rng_pool);
+
+  std::string swarm_name = "my particles";
+  Metadata swarm_metadata;
+  pkg->AddSwarm(swarm_name, swarm_metadata);
+  Metadata real_swarmvalue_metadata({Metadata::Real});
+  pkg->AddSwarmValue("vx", swarm_name, real_swarmvalue_metadata);
+  pkg->AddSwarmValue("vy", swarm_name, real_swarmvalue_metadata);
+  pkg->AddSwarmValue("vz", swarm_name, real_swarmvalue_metadata);
+  pkg->AddSwarmValue("weight", swarm_name, real_swarmvalue_metadata);
+
+  std::string field_name = "particle_deposition";
+  Metadata m({Metadata::Cell, Metadata::Independent});
+  pkg->AddField(field_name, m);
+
+  pkg->EstimateTimestep = EstimateTimestep;
+
+  return pkg;
+}
+
+AmrTag CheckRefinement(Container<Real> &rc) { return AmrTag::same; }
+
+Real EstimateTimestep(std::shared_ptr<Container<Real>> &rc) {
+  //auto pmb = rc->pmy_block;
```

Comment:
Remove this.

Reply:
Good catch, thanks (similar issue in next function)